dataland-frontend/src/components/forms/parts/UploadReports.vue

```diff
@@ -60,49 +64,74 @@
 <script lang="ts">
 import { defineComponent } from "vue";
 import PrimeButton from "primevue/button";
-import { CompanyReport } from "@clients/backend";
 import ReportFormElement from "@/components/forms/parts/ReportFormElement.vue";
 import ElementsDialog from "@/components/general/ElementsDialog.vue";
 import { ReportToUpload, StoredReport } from "@/utils/FileUploadUtils";
 import UploadDocumentsForm from "@/components/forms/parts/elements/basic/UploadDocumentsForm.vue";
+import { YesNoFormFieldProps } from "@/components/forms/parts/fields/FormFieldProps";
+import { CompanyReport } from "@clients/backend";
+import { ObjectType } from "@/utils/UpdateObjectUtils";
 
 export default defineComponent({
   name: "UploadReports",
+  inheritAttrs: false,
+  inject: {
+    injectReferencedReportsForPrefill: {
+      from: "referencedReportsForPrefill",
+      default: {},
+    },
+  },
   components: {
     UploadDocumentsForm,
     ReportFormElement,
     PrimeButton,
   },
-  emits: ["referenceableReportNamesChanged"],
+  emits: ["reportsUpdated"],
   data() {
     return {
       reportsToUpload: [] as ReportToUpload[] | undefined,
       storedReports: [] as StoredReport[],
     };
   },
   props: {
+    ...YesNoFormFieldProps,
     referencedReportsForPrefill: {
       type: Object as () => { [key: string]: CompanyReport },
     },
-  },
-  computed: {
-    allReferenceableReportNames(): string[] {
-      const namesOfFilesToUpload = this.reportsToUpload.map((reportToUpload) => reportToUpload.fileNameWithoutSuffix);
-      const namesOfStoredReports = this.storedReports.map((storedReport) => storedReport.reportName);
-      return namesOfFilesToUpload.concat(namesOfStoredReports);
+    isEuTaxonomy: {
+      type: Boolean,
+      default: false,
     },
   },
   watch: {
     referencedReportsForPrefill() {
       this.prefillAlreadyUploadedReports();
     },
   },
+  computed: {
+    namesOfFilesToUpload() {
+      return this.reportsToUpload.map((reportToUpload) => reportToUpload.fileNameWithoutSuffix);
+    },
+    namesOfStoredReports() {
+      return this.storedReports.map((storedReport) => storedReport.reportName);
+    },
+    allReferenceableReportNames(): string[] {
+      return this.namesOfFilesToUpload.concat(this.namesOfStoredReports);
+    },
+  },
+  mounted() {
+    this.prefillAlreadyUploadedReports();
+  },
   methods: {
     /**
      * Emits event that referenceable files changed
      */
     emitReferenceableReportNamesChangedEvent() {
-      this.$emit("referenceableReportNamesChanged", this.allReferenceableReportNames);
+      if (this.isEuTaxonomy) {
+        this.$emit("reportsUpdated", this.allReferenceableReportNames);
+      } else {
+        this.$emit("reportsUpdated", this.allReferenceableReportNames, this.reportsToUpload);
```

Comment:
I like the approach for SFDR: You emit the reports to the parent, and the parent saves them in "documents". Then it uploads everything in "documents" when submit is clicked.

Question @MarcinStanczyk7 :   Could you also do this approach for EuTaxonomy?
If it is very much effort, give up. But if not, this would be very good.

If you manage to do that, we can always emit `reportsToUpload`.

Reply:
Should we postpone this, as this PR is already very long and deadline day is coming closer and we will redo part of eu taxonomy anyways so we could do this then?

Reply:
If Marcin thinks this takes more than 15min, then let's not do this.
Otherwise let's do this, since this is really confusing for outsiders.

Reply:
Let's finish Sfdr as it will be merged to main. We will improve EuTaxonomy ?

Reply:
Ok 😢